tests/Integration/Behaviour/Features/Context/AbstractPrestaShopFeatureContext.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+namespace Tests\Integration\Behaviour\Features\Context;
+
+use Behat\Behat\Context\Context as BehatContext;
+use LegacyTests\PrestaShopBundle\Utils\DatabaseCreator;
+
+/**
+ * PrestaShopFeatureContext provides behat hooks to perform necessary operations for testing:
+ * - shop setup
+ * - database reset between scenario
+ * - cache clear between steps
+ * - ...
+ */
+abstract class AbstractPrestaShopFeatureContext implements BehatContext
+{
+    /**
+     * PrestaShop Symfony AppKernel
+     *
+     * Required to access services through the container
+     *
+     * @var \AppKernel
+     */
+    protected static $kernel;
+
+    /**
+     * "When" steps perform actions, and some of them store the latest result
+     * in this variable so that "Then" action can check its content
+     *
+     * @var mixed
+     */
+    protected $latestResult;
+
+    /** @var bool */
+    protected $flag_performDatabaseCleanHard = false;
+    /** @var bool */
+    protected $flag_performDatabaseCleanLight = false;
+
+    /**
+     * @BeforeSuite
+     */
+    public static function prepare($scope)
+    {
+        require_once __DIR__ . '/bootstrap.php';
+
+        self::$kernel = new \AppKernel('test', true);
+        self::$kernel->boot();
+    }
+
+    /**
+     * This hook can be used to flag a scenario as eligible for database hard reset
+     *
+     * @BeforeScenario @database-hard-reset
+     */
+    public function beforeScenario_cleanDatabaseHard()
+    {
+        $this->flag_performDatabaseCleanHard = true;
+    }
+
+    /**
+     * This hook can be used to flag a scenario as eligible for database light reset
+     *
+     * @BeforeScenario @database-light-reset
+     */
+    public function beforeScenario_cleanDatabaseLight()
+    {
+        $legacyDatabaseSingleton = \Db::getInstance(_PS_USE_SQL_SLAVE_);
```

Comment:
Prefer to use namespace, it will be easier to change a one line at the top of the file instead of many occurrences.
Same for AppKernel please.